plugins/services/src/js/components/ServiceStatusProgressBar.tsx

```diff
@@ -45,14 +45,9 @@ class ServiceStatusProgressBar extends React.Component<
   render() {
     const { service } = this.props;
     const instancesCount = service.getInstancesCount();
-    const serviceStatus = service.getServiceStatus();
     const runningInstances = service.getRunningInstancesCount();
 
-    if (
-      serviceStatus === ServiceStatus.RUNNING ||
-      serviceStatus === ServiceStatus.STOPPED ||
-      serviceStatus === ServiceStatus.NA
-    ) {
+    if (!ServiceStatus.showProgressBar(service.getServiceStatus())) {
       return null;
```

Comment:
We are changing the logic here. It used to be a positive check and now it is negative. This might be relevant if we add more statuses in the future.

Reply:
That's intentional. Would you feel more comfortable with explicitly mapping each Status `showProgressBar` to whether is showsProgress or not?

Reply:
Before we didn't show progress bar if it was "running", "stopped" or "na". Now, we show it if its' https://github.com/dcos/dcos-ui/pull/3748/files#diff-850dfaf716778762300d8d5407693b70R154 . We changed from checking what it wasn't to checking what it is.

Reply:
Yes indeed, that's intentional as we have a fixed set of statuses - with the set of statuses that have a progressBar being far smaller. you now need to opt in to have a statusBar with your status.

Would you feel more comfortable with explicitly mapping Statuses to whether they show a progressBar, e.g.

```
const STOPPED: Status = {
  priority: 2,
  displayName: i18nMark("Stopped"),
  icon: StatusIcon.STOPPED,
  showProgressBar: false
};
```
so that when someone introduces a new Status they are forced to decide on that?

Reply:
Which statuses do actually show a progress bar and which don't?

Reply:
Resolved in chat. The formerly shown are the ones in `ServiceStatus.showProgressBar`.